### PR TITLE
Move LockDiff from dep into gps

### PR DESCRIPTION
**Pull request by carolynvs**
* Move `LockDiff` and related structs.
* Expose `gps.DiffLocks(l1, l2)` and `gps.DiffProjects(p1, p2)`.
* All the TOML related annotation will remain in dep, using the same `rawX` pattern that locks use.

### CODE REVIEW

File: lockdiff.go [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+
+// DiffLocks compares two locks and identifies the differences between them.
+// Returns nil if there are no differences.
+func DiffLocks(l1 Lock, l2 Lock) *LockDiff {
+	// Default nil locks to empty locks, so that we can still generate a diff
+	if l1 == nil {
+		l1 = &SimpleLock{}
+	}
+	if l2 == nil {
+		l2 = &SimpleLock{}
```
Comment (sdboyer) [inline lockdiff.go]: Let's get some simple coverage for the zero value case

File: version.go [hunk 2]
```diff
@@ -810,3 +810,23 @@ func hidePair(pvl []PairedVersion) []Version {
 	}
 	return vl
 }
+
+// Decompose a Version into the underlying number, branch and revision
```
Comment (sdboyer) [inline version.go]: Let's rename this to something a little more descriptive, if we're gonna export it - `Info` is pretty bland 😄  

how about `VersionComponentStrings()`?

**Comment by carolynvs** [comment]
I've renamed it per your suggestion and added tests for nil/empty locks.

**Comment by sdboyer** [comment]
Great - in we go! Thanks! 🎉

**Comment by carolynvs** [comment]
In before reorg! 😁

**Comment by sdboyer** [comment]
🏁 🏁 🏁

**Pull request merged**